models/llama3/reference_impl/generation.py

```diff
@@ -107,8 +112,10 @@ def build(
                 model_parallel_size = int(os.environ.get("WORLD_SIZE", 1))
             initialize_model_parallel(model_parallel_size)
 
-        local_rank = int(os.environ.get("LOCAL_RANK", 0))
-        torch.cuda.set_device(local_rank)
+        
+        if _is_cuda(device):
+            local_rank = int(os.environ.get("LOCAL_RANK", 0))
+            torch.cuda.set_device(local_rank)
```

Comment:
`local_rank` must be defined for all code branches otherwise execution will fail on line 122 below (`if local_rank > 0`)

```suggestion
        local_rank = int(os.environ.get("LOCAL_RANK", 0))
        if _is_cuda(device):
            torch.cuda.set_device(local_rank)
```